src/Surfnet/StepupMiddleware/ApiBundle/Controller/InstitutionConfigurationOptionsController.php

```diff
@@ -26,21 +26,32 @@
 
 final class InstitutionConfigurationOptionsController extends Controller
 {
-    public function getForInstitutionAction($institution)
+    public function getForInstitutionAction($institutionValue)
     {
         $this->denyAccessUnlessGranted(['ROLE_SS', 'ROLE_RA']);
 
+        $institution = new Institution($institutionValue);
+
         $institutionConfigurationOptions = $this
             ->getInstitutionConfigurationOptionsService()
-            ->findInstitutionConfigurationOptionsFor(new Institution($institution));
+            ->findInstitutionConfigurationOptionsFor($institution);
+
+        $allowedSecondFactorList = $this
+            ->getAllowedSecondFactorListService()
+            ->getAllowedSecondFactorListFor($institution);
 
         if ($institutionConfigurationOptions === null) {
             throw new NotFoundHttpException(
-                sprintf('No institution configuration options found for institution "%s"', $institution)
+                sprintf('No institution configuration options found for institution "%s"', $institutionValue)
             );
         }
 
-        return new JsonResponse($institutionConfigurationOptions);
+        return new JsonResponse([
```

Comment:
hrms, this is a shame... and necessary it seems. We should remove the JsonSerializable from the InstitutionConfigurationOptions though, to prevent assumptions of devs.